cmd/vat/evaluation/evaluator.go

```diff
@@ -0,0 +1,130 @@
+package evaluation
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/elrond-go/cmd/vat/analysis"
+	"github.com/elrond-go/cmd/vat/scan"
+	"github.com/elrond-go/cmd/vat/utils"
+)
+
+type EvaluatedTarget struct {
+	Address        string
+	Ports          []scan.Port
+	Status         string
+	Score          int
+	SecurityLevel  utils.SecureLevel
+	Judgements     []string
+	EvaluationType utils.EvaluationType
+	scannerFactory analysis.ScannerFactory
+}
+
+func (eT *EvaluatedTarget) Evaluate() EvaluatedTarget {
+	eT.Score -= eT.calculateDeductionPoints()
+	eT.SecurityLevel = eT.calculateSecurityLevel()
+	eT.Status = string(utils.EVALUATED)
+	return *eT
+}
+
+func (eT *EvaluatedTarget) calculateDeductionPoints() (deductionPoints int) {
+	judgement := utils.JudgementFromPort
+	deductionPoints = 0
+
+	for _, port := range eT.Ports {
+		if port.State == utils.Open {
+			if eT.requestedSpecialCheck(port.Number) {
+				judgement = eT.runSpecialCheck()
+			} else {
+				judgement = utils.JudgementFromPort
+			}
+			deductionPoints += eT.runJudgement(port.Importance, judgement)
+		}
+	}
+
+	return
+}
+
+func (eT *EvaluatedTarget) requestedSpecialCheck(portNumber int) bool {
+	return ((portNumber == 22) && (eT.EvaluationType == utils.Polite_PortAndSshEvaluation)) ||
+		((portNumber == 22) && (eT.EvaluationType == utils.Brute_PortAndSshEvaluation))
+}
+
+func (eT *EvaluatedTarget) runSpecialCheck() (judgement utils.Judgement) {
+	if eT.EvaluationType == utils.Polite_PortAndSshEvaluation {
+		return eT.runPoliteSshCheck()
+	} else {
+		return eT.runBruteForceSshCheck()
+	}
+}
+
+func (eT *EvaluatedTarget) runPoliteSshCheck() (judgement utils.Judgement) {
+	s := eT.scannerFactory.CreateScanner(eT.Address, utils.TCP_POLITE_REQ1)
+
+	_, err := s.Scan()
+	if err != nil {
+		return interpretErrorGravity(err)
+	}
+
+	return utils.JudgementDummyPermited
+}
+
+func interpretErrorGravity(err error) (judgement utils.Judgement) {
+	errSlice := strings.Split(err.Error(), " ")
+
+	if errSlice[9] == "[none]," {
```

Comment:
magic string. What "[none],", "password]," should mean. Also, hardcoded index 9. What does 9 mean? Should have been 8 or 10?

Reply:
fixed this... somehow...
ssh.Dial() returns an error as a string: _"[DEBUG] SSH handshake err: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none password], no supported methods remain"_ therefore the split of the string.

```suggestion
		for i := range errSlice {
		if errSlice[i] == "[none]," {
			return utils.JudgementSshUserPermited
		}
```